compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -102,21 +107,44 @@ upgradeTemplate n =
   , "                    create $ conv d"
   ]
 
+-- | Generate the full source for a daml-lf package.
+generateSrcPkgFromLf ::
+       LF.PackageId
+    -> MS.Map GHC.UnitId LF.PackageId
+    -> LF.Package
+    -> [(NormalizedFilePath, String)]
+generateSrcPkgFromLf thisPkgId pkgMap pkg = do
+    mod <- NM.toList $ LF.packageModules pkg
+    let fp =
+            toNormalizedFilePath $
```

Comment:
Looks like a reasonable function to factor out into `daml-lf-tools`.

Reply:
I'll do a bigger refactoring of everything once everything works smoothly.